ktor-client/ktor-client-core/common/src/io/ktor/client/features/HttpRedirect.kt

```diff
@@ -32,15 +34,14 @@ class HttpRedirect {
             while (true) {
                 val location = call.response.headers[HttpHeaders.Location]
 
-                call.close()
-
-                call = execute(HttpRequestBuilder().apply {
+                val requestBuilder = HttpRequestBuilder().apply {
                     takeFrom(origin.request)
                     url.parameters.clear()
 
                     location?.let { url.takeFrom(it) }
-                })
+                }
 
+                call = execute(requestBuilder)
                 if (!call.response.status.isRedirect()) return call
```

Comment:
I'm just curious, what if there is a redirect response with body, what would happend and who does consume/discard it here?

Reply:
We handle it in the send feature